File: Arquitectura_G04/proyecto_conjunta/restfull_java_pruebaproyecto/banquito-server/src/main/java/ec/edu/banquito/entity/CuotaCredito.java
package ec.edu.banquito.entity;

import jakarta.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "CUOTA_CREDITO")
public class CuotaCredito implements Serializable {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cuota")
    private Integer idCuota;
    
    @ManyToOne
    @JoinColumn(name = "id_credito", nullable = false)
    private Credito credito;
    
    @Column(name = "numero_cuota", nullable = false)
    private Integer numeroCuota;
    
    @Column(name = "fecha_vencimiento", nullable = false)
    private LocalDate fechaVencimiento;
    
    @Column(name = "valor_cuota", nullable = false, precision = 18, scale = 2)
    private BigDecimal valorCuota;
    
    @Column(name = "interes_pagado", nullable = false, precision = 18, scale = 2)
    private BigDecimal interesPagado;
    
    @Column(name = "capital_pagado", nullable = false, precision = 18, scale = 2)
    private BigDecimal capitalPagado;
    
    @Column(name = "saldo_restante", nullable = false, precision = 18, scale = 2)
    private BigDecimal saldoRestante;
    
    public CuotaCredito() {}
    
    // Getters y Setters
    public Integer getIdCuota() {
        return idCuota;
    }
    
    public void setIdCuota(Integer idCuota) {
        this.idCuota = idCuota;
    }
    
    public Credito getCredito() {
        return credito;
    }
    
    public void setCredito(Credito credito) {
        this.credito = credito;
    }
    
    public Integer getNumeroCuota() {
        return numeroCuota;
    }
    
    public void setNumeroCuota(Integer numeroCuota) {
        this.numeroCuota = numeroCuota;
    }
    
    public LocalDate getFechaVencimiento() {
        return fechaVencimiento;
    }
    
    public void setFechaVencimiento(LocalDate fechaVencimiento) {
        this.fechaVencimiento = fechaVencimiento;
    }
    
    public BigDecimal getValorCuota() {
        return valorCuota;
    }
    
    public void setValorCuota(BigDecimal valorCuota) {
        this.valorCuota = valorCuota;
    }
    
    public BigDecimal getInteresPagado() {
        return interesPagado;
    }
    
    public void setInteresPagado(BigDecimal interesPagado) {
        this.interesPagado = interesPagado;
    }
    
    public BigDecimal getCapitalPagado() {
        return capitalPagado;
    }
    
    public void setCapitalPagado(BigDecimal capitalPagado) {
        this.capitalPagado = capitalPagado;
    }
    
    public BigDecimal getSaldoRestante() {
        return saldoRestante;
    }
    
    public void setSaldoRestante(BigDecimal saldoRestante) {
        this.saldoRestante = saldoRestante;
    }
}

File: Arquitectura_G04/proyecto_conjunta/soap_dotnet_pruebaproyecto/BancoSoapService/Models/CuotaCredito.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BancoSoapService.Models
{
    [Table("CUOTA_CREDITO")]
    public class CuotaCredito
    {
        [Key]
        [Column("id_cuota")]
        public int IdCuota { get; set; }

        [Required]
        [Column("id_credito")]
        public int IdCredito { get; set; }

        [Required]
        [Column("numero_cuota")]
        public int NumeroCuota { get; set; }

        [Required]
        [Column("fecha_vencimiento")]
        public DateTime FechaVencimiento { get; set; }

        [Required]
        [Column("valor_cuota", TypeName = "decimal(18,2)")]
        public decimal ValorCuota { get; set; }

        [Required]
        [Column("interes_pagado", TypeName = "decimal(18,2)")]
        public decimal InteresPagado { get; set; }

        [Required]
        [Column("capital_pagado", TypeName = "decimal(18,2)")]
        public decimal CapitalPagado { get; set; }

        [Required]
        [Column("saldo_restante", TypeName = "decimal(18,2)")]
        public decimal SaldoRestante { get; set; }

        // Navegación
        [ForeignKey("IdCredito")]
        public virtual Credito? Credito { get; set; }
    }
}
